lib/stream/xlsx/worksheet-writer.js

```diff
@@ -450,6 +451,39 @@ class WorksheetWriter {
     return this._background && this._background.imageId;
   }
 
+  // =========================================================================
+  // Worksheet Protection
+  protect(password, options) {
+    // TODO: make this function truly async
+    // perhaps marshal to worker thread or something
+    return new Promise(resolve => {
```

Comment:
Whether this attribute should be declared in `constructor ()`.
```javascript
    // worksheet protection
    this.sheetProtection = null;
```

Reply:
This has been added to the constructor

Reply:
Does this really work ? I set sheet protection to true it doesn't lock a particular cell